pkg/actions/cluster/unowned.go

```diff
@@ -1,23 +1,236 @@
+	"github.com/aws/aws-sdk-go/aws/awserr"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/weaveworks/eksctl/pkg/utils/tasks"
+
+	"github.com/pkg/errors"
+
+	"github.com/weaveworks/eksctl/pkg/utils/waiters"
+
+	"github.com/weaveworks/eksctl/pkg/kubernetes"
+
+	iamoidc "github.com/weaveworks/eksctl/pkg/iam/oidc"
+
+	"github.com/kris-nova/logger"
+
+	awseks "github.com/aws/aws-sdk-go/service/eks"
+
 	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
 	"github.com/weaveworks/eksctl/pkg/eks"
+	kubeclient "k8s.io/client-go/kubernetes"
 )
 
 type UnownedCluster struct {
-	cfg *api.ClusterConfig
-	ctl *eks.ClusterProvider
+	cfg       *api.ClusterConfig
+	ctl       *eks.ClusterProvider
+	clientSet kubeclient.Interface
 }
 
-func NewUnownedCluster(cfg *api.ClusterConfig, ctl *eks.ClusterProvider) (*UnownedCluster, error) {
+func NewUnownedCluster(cfg *api.ClusterConfig, ctl *eks.ClusterProvider, clientSet kubeclient.Interface) *UnownedCluster {
 	return &UnownedCluster{
-		cfg: cfg,
-		ctl: ctl,
-	}, nil
+		cfg:       cfg,
+		ctl:       ctl,
+		clientSet: clientSet,
+	}
 }
 
 func (c *UnownedCluster) Upgrade(dryRun bool) error {
 	_, err := upgrade(c.cfg, c.ctl, dryRun)
 	return err
 }
+
+func (c *UnownedCluster) Delete(waitInterval time.Duration, wait bool) error {
+	clusterName := c.cfg.Metadata.Name
+
+	if err := c.checkClusterExists(clusterName); err != nil {
+		return err
+	}
+
+	if err := deleteSharedResources(c.cfg, c.ctl, c.clientSet); err != nil {
+		return err
+	}
+
+	// we have to wait for nodegroups to delete before deleting the cluster
+	// so the `wait` value is ignored here
+	if err := c.deleteAndWaitForNodegroupsDeletion(clusterName, c.ctl.Provider.WaitTimeout(), waitInterval); err != nil {
+		return err
+	}
+
+	if err := c.deleteIAMAndOIDC(wait, kubernetes.NewCachedClientSet(c.clientSet)); err != nil {
+		return err
+	}
+
+	err := c.deleteCluster(clusterName, c.ctl.Provider.WaitTimeout(), wait)
+	if err != nil {
+		return err
+	}
+
+	logger.Success("all cluster resources were deleted")
+	return nil
+}
+
+func (c *UnownedCluster) checkClusterExists(clusterName string) error {
+	_, err := c.ctl.Provider.EKS().DescribeCluster(&awseks.DescribeClusterInput{
+		Name: &c.cfg.Metadata.Name,
+	})
+	if err != nil {
+		if isNotFound(err) {
+			return errors.Errorf("cluster %q not found", clusterName)
+		}
+		return errors.Wrapf(err, "error describing cluster %q", clusterName)
+	}
+	return nil
+}
+
+func (c *UnownedCluster) deleteIAMAndOIDC(wait bool, clientSetGetter kubernetes.ClientSetGetter) error {
+	clusterOperable, _ := c.ctl.CanOperate(c.cfg)
+
+	var oidc *iamoidc.OpenIDConnectManager
+	var err error
+	stackManager := c.ctl.NewStackManager(c.cfg)
+
+	oidcSupported := true
+	if clusterOperable {
+		oidc, err = c.ctl.NewOpenIDConnectManager(c.cfg)
+		if err != nil {
+			if _, ok := err.(*eks.UnsupportedOIDCError); !ok {
+				return err
+			}
+			oidcSupported = false
+		}
+	}
+
+	tasksTree := &tasks.TaskTree{Parallel: false}
```

Comment:
```suggestion
      var oidc *iamoidc.OpenIDConnectManager

      clusterOperable, _ := c.ctl.CanOperate(c.cfg)
      oidcSupported := true

      if clusterOperable {
            var err error
            oidc, err = c.ctl.NewOpenIDConnectManager(c.cfg)
            if err != nil {
                  if _, ok := err.(*eks.UnsupportedOIDCError); !ok {
                        return err
                  }
                  oidcSupported = false
            }
      }

      tasksTree := &tasks.TaskTree{Parallel: false}
      stackManager := c.ctl.NewStackManager(c.cfg)
```

mini scoping nit (diff annoying looks bigger than it is)